sentry_sdk/scope.py

```diff
@@ -147,21 +187,145 @@ class Scope(object):
         "_force_auto_session_tracking",
         "_profile",
         "_propagation_context",
+        "client",
+        "original_scope",
+        "_ty",
     )
 
-    def __init__(self):
-        # type: () -> None
+    def __init__(self, ty=None, client=None):
+        # type: (Optional[str], Optional[sentry_sdk.Client]) -> None
+        self._ty = ty
+        self.original_scope = None  # type: Optional[Scope]
+
         self._event_processors = []  # type: List[EventProcessor]
         self._error_processors = []  # type: List[ErrorProcessor]
 
         self._name = None  # type: Optional[str]
         self._propagation_context = None  # type: Optional[Dict[str, Any]]
 
+        self.client = (
+            NoopClient()
+        )  # type: Union[sentry_sdk.Client, sentry_sdk.client.NoopClient]
```

Comment:
Since Client inherits from NoopClient, isn't this union redundant?

Reply:
fixed.